File: 121 Best Time to Buy and Sell Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        if(prices == null || prices.length == 0) return 0;
        int max = 0, min = prices[0];
        int profit = 0;
        for (int i = 1; i < prices.length; i++) {
            if(prices[i] < min) {
                min = prices[i]; // Change the minimum
            } else {
                if(prices[i] - min > profit) {
                    profit = prices[i] - min; // Check the profit
                }
                    
            }
        }
        return profit;
    }
}

File: 121 Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if not prices: return 0
        res, low = 0, prices[0]
        for i in range(1, len(prices)):
            if prices[i] < low:
                low = prices[i]
            else:
                res = max(res, prices[i] - low)
        return res